drivers/counter/counter_mcux_rtc.c

```diff
@@ -236,6 +236,17 @@ static int mcux_rtc_init(const struct device *dev)
 	RTC_GetDefaultConfig(&rtc_config);
 	RTC_Init(config->base, &rtc_config);
 
+	BUILD_ASSERT(((strcmp(DT_INST_PROP(0, clock_source), "LPO") == 0) &&
+		FSL_FEATURE_RTC_HAS_LPO_ADJUST) ||
+		(strcmp(DT_INST_PROP(0, clock_source), "RTC") == 0),
+		"Cannot choose the LPO clock for that instance of the RTC");
```

Comment:
this is just a question, is `strcmp` actually tested at build time?

Reply:
In this case, the strcmp function is used as part of the BUIlD_ASSERT condition, which performs static analysis and calculates the string result at build time

Reply:
![image](https://github.com/user-attachments/assets/dcb4cb05-f6a6-4e58-b8a5-b5a06c291be3)

this does not look like it would be computed at build time